core/commands/storage/upload/upload/renew.go

```diff
@@ -0,0 +1,253 @@
+`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("file-hash", true, false, "Hash of file to renew."),
+		cmds.StringArg("renew-length", true, false, "New File storage period on hosts in days."),
+	},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		n, err := cmdenv.GetNode(env)
+		if err != nil {
+			return err
+		}
+		cfg, err := n.Repo.Config()
+		if err != nil {
+			return err
+		}
+		fileHash := req.Arguments[0]
+		renterId := n.Identity.String()
+		metaReq := &guardpb.CheckFileStoreMetaRequest{
+			FileHash:     fileHash,
+			RenterPid:    renterId,
+			RequesterPid: renterId,
+			RequestTime:  time.Now().UTC(),
+		}
+		sig, err := crypto.Sign(n.PrivateKey, metaReq)
+		if err != nil {
+			return err
+		}
+		metaReq.Signature = sig
+		ctx, _ := helper.NewGoContext(req.Context)
+		var meta *guardpb.FileStoreStatus
+		err = grpc.GuardClient(cfg.Services.GuardDomain).WithContext(ctx, func(ctx context.Context,
+			client guardpb.GuardServiceClient) error {
+			meta, err = client.CheckFileStoreMeta(ctx, metaReq)
+			if err != nil {
+				return err
+			}
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+		contracts := meta.Contracts
+		if len(contracts) <= 0 {
+			return errors.New("length of contracts is 0")
+		}
+		ssId, _ := uh.SplitContractId(contracts[0].ContractId)
+
+		shardIndexes := make([]int, 0)
+		shardHashes := make([]string, 0)
+
+		for _, contract := range contracts {
+			if contract.State == guardpb.Contract_UPLOADED {
+				shardHashes = append(shardHashes, contract.ShardHash)
+				shardIndexes = append(shardIndexes, int(contract.ShardIndex))
+			}
+		}
+
+		renewPeriod, err := strconv.ParseInt(req.Arguments[1], 10, 0)
+		if err != nil {
+			return err
+		}
+
+		ctxParams, err := uh.ExtractContextParams(req, env)
+		if err != nil {
+			return err
+		}
+		rss, err := sessions.GetRenterSession(ctxParams, ssId, fileHash, shardHashes)
+		if err != nil {
+			return err
+		}
+
+		m := contracts[0].ContractMeta
+		renterPid, err := peer.IDB58Decode(renterId)
+		if err != nil {
+			return err
+		}
+
+		var price int64
+		err = grpc.HubQueryClient(cfg.Services.HubDomain).WithContext(ctx,
+			func(ctx context.Context, client hubpb.HubQueryServiceClient) error {
+				req := new(hubpb.SettingsReq)
+				//unknow host id
+				req.Id = ""
+				resp, err := client.GetSettings(ctx, req)
+				if err != nil {
+					return err
+				}
+				if resp.Code != hubpb.ResponseCode_SUCCESS {
+					return errors.New(resp.Message)
+				}
+				price = int64(resp.SettingsData.StoragePriceAsk)
+				return nil
+			})
+		if err != nil {
+			return err
+		}
+		//Build new contracts here
+		CreateNewContracts(rss, price, m.ShardFileSize, m.RentEnd, int(renewPeriod), false, renterPid, -1,
+			shardIndexes, nil)
+
+		seRes := &Res{
+			ID: ssId,
+		}
+		return res.Emit(seRes)
+	},
+	Type: Res{},
+}
+
+func CreateNewContracts(rss *sessions.RenterSession, price int64, shardSize int64, preRentEnd time.Time,
```

Comment:
Merge with `func UploadShard` in upload_shard.go. Otherwise it's hard to maintain two similar upload routines.